app/src/main/java/com/github/se/cyrcle/model/review/ReviewViewModel.kt

```diff
@@ -268,6 +283,32 @@ class ReviewViewModel(
     Log.d("ReviewViewModel", "Review and metrics updated successfully: ${selectedReview.nbReports}")
   }
 
+  fun addReportingUser(user: User) {
+    _selectedReview.update { currentReview ->
+      currentReview?.let {
+        val updatedReview = it.copy(reportingUsers = it.reportingUsers + user.public.userId)
+
+        reviewRepository.updateReview(
+            review = updatedReview,
+            onSuccess = {
+              Log.d("ReviewViewModel", "User added to reportingUsers in Firestore")
+              _selectedReview.update { state ->
```

Comment:
Maybe I'm mistaken here, but I don't think that `state` is a good name.
The below `if (...) updatedReview else state` feels really weird to read, so I would go for ~`state`~ to `currentReview`.
Also, should it be `if (...) { update{} }` instead of ~`update { if (...) {} else {} }`~, because the current implementation might trigger some listener chain for nothing, hence why I would think to put the `if` outside of this `update` lambda.

Reply:
I've spent a few hours trying to get an alternative to work and it unfortunately hasn't as of yet. I've reverted this branch to the state it was at before I addressed your change. Is this really blocking for you? I think, given the stateFlows that I'm using that the permanent update is necessary, at least that's what my tests strongly suggest